agent.py

```diff
@@ -602,8 +602,8 @@ def parent_func():
                     check.stop()
 
     elif 'configcheck' == command or 'configtest' == command:
-        configcheck()
-        sd_configcheck(agentConfig)
+        if configcheck() or sd_configcheck(agentConfig):
```

Comment:
`sd_configcheck` doesn't return anything so I don't think it makes sense to have it in the condition here. I'm not sure we could make that function return a sensible value that reflects the status of the service discovery confs...

So instead of this, I think we should simply make this code return the value that `configcheck` returns.  The part of the code under `__main__` will then exit with the correct status code.